oci/spec_opts.go

```diff
@@ -1264,3 +1273,32 @@ func WithDevShmSize(kb int64) SpecOpts {
 		return ErrNoShmMount
 	}
 }
+
+// freebsdLinuxEmulationMounts modifies the mount spec to mount emulated Linux filesystems on FreeBSD,
+// as per: https://wiki.freebsd.org/LinuxJails
+func freebsdLinuxEmulationMounts(s *Spec) error {
+	var mounts []specs.Mount
+	for _, m := range s.Mounts {
+		path := filepath.Clean(m.Destination)
+		if path == "/proc" {
+			mounts = append(mounts, specs.Mount{
+				Destination: "/proc",
+				Type:        "linprocfs",
+				Source:      "linprocfs",
+				Options:     []string{},
+			})
```

Comment:
What's the reasoning behind only adding `linprocfs` if a `/proc` mount is already defined in the spec?  Wouldn't we always want to append this if we're adding Linux emulation?

Reply:
I figured to not add mounts were not already defined, besides the linux specific ones. Making sure it's always there is also fine for me ( I guess Linux always likes to have it?).

I'll update tonight.